### PR TITLE
move rust workflow to use nix environment

### PR DESCRIPTION
**Pull request by metaclips**
This PR moves our Rust workflows to use the Nix environment

### CODE REVIEW

File: implementations/rust/Makefile [hunk 1]
```diff
@@ -1,37 +1,82 @@
...
-	cargo build --package $*
+	cargo build --locked --package $*
+
+check_no_std:
+	cargo --version
+	cargo check --locked --target thumbv7em-none-eabihf -p ockam --no-default-features --features 'no_std alloc software_vault'
+	# no_std example project
+	cd examples/rust/example_projects/no_std
+	cargo check --example hello
+	# RUSTFLAGS='-Dwarnings' cargo check --example hello --no-default-features --features 'alloc no_std'
```
Comment (etorreborre) [inline implementations/rust/Makefile]: Can you please remove the commented out code here?

**Review by etorreborre — DISMISSED** [review]
This looks good! I wanted to try some commands locally and couldn't find a way to execute them. Could you please provide an example in the README?

**Review by etorreborre — DISMISSED** [review]
Great stuff and thanks for the README!

As suggested before I think it would eventually be a great idea to switch to `just` instead of `make`: https://github.com/casey/just

**Comment by metaclips** [comment]
> Great stuff and thanks for the README!
> 
> As suggested before I think it would eventually be a great idea to switch to `just` instead of `make`: https://github.com/casey/just

Thanks @etorreborre we can treat adding Just tooling in subsequent PRs

**Review by etorreborre — DISMISSED** [review]

**Comment by metaclips** [comment]
@etorreborre there's an inconsistency in the merge queue caused by this PR, I am pushing a fix

**Review by etorreborre — DISMISSED** [review]

**Review by etorreborre — APPROVED** [review]

**Pull request merged**